### PR TITLE
layers: Update to 1.4

### PR DESCRIPTION
**Pull request by mikes-lunarg**
This is @arno-lunarg's branch rebased onto main with the following changes:
- Tweaked known_good.json to point to public 1.4 dependencies
- Regenerated code
- Removed an infinite recursion in `PostCallRecordCmdPushConstants2KHR`
- Resolved merge conflicts with #8950 in `cc_copy_blit_resove.cpp`

### CODE REVIEW

**Comment by spencer-lunarg** [comment]
@arno-lunarg we hit these errors CI is hitting, we had a fix I thought

File: layers/VkLayer_khronos_validation.json.in [hunk 1]
```diff
@@ -1057,15 +1057,30 @@
...
-                    "key": "duplicate_message_limit",
-                    "env": "VK_LAYER_DUPLICATE_MESSAGE_LIMIT",
-                    "label": "Max Duplicated Messages",
-                    "description": "Maximum number of times any single validation message should be reported. Value of zero will print forever",
-                    "type": "INT",
-                    "default": 10,
-                    "range": {
-                        "min": 0
-                    }
+                    "key": "enable_message_limit",
```
Comment (spencer-lunarg) [inline layers/VkLayer_khronos_validation.json.in:1060]: not sure why this is part of this PR

**Review by spencer-lunarg — APPROVED** [review]
I'm happy if CI is happy, we will have cleanup after this PR I assumed as well

File: layers/VkLayer_khronos_validation.json.in (hunk 1 above)
Comment (arno-lunarg) [reply layers/VkLayer_khronos_validation.json.in:1060]: Ah it probably got through merge conflicts hell

**Comment by arno-lunarg** [comment]
There are some VUs oddities: `VUID-VkBindDescriptorSetsInfo-layout-parameter` is now part of the spec, but I still only see `VUID-VkBindDescriptorSetsInfo-layout-parameter` in `generated\vk_validation_error_messages.h`

**Comment by spencer-lunarg** [comment]
Created an issue here - https://github.com/KhronosGroup/Vulkan-Headers/issues/517

for now the `validusage.json` being wrong should not delay this from getting merged, it will make the `vk_validation_stats.py` script cry, but that is just for us and can be resolved after getting the headers in

**Comment by arno-lunarg** [comment]
Well VUs are missing, so some auto generated validation is missing, and tests fail because of that

**Comment by spencer-lunarg** [comment]
> Well VUs are missing, so some auto generated validation is missing, and tests fail because of that

aww, well let me try to make a PR to the Headers to get the new `validusage.json` in to unblock this... I rather not have to undo the VUID and then redo them

**Comment by arno-lunarg** [comment]
Thank you <3

File: scripts/known_good.json [hunk 2]
```diff
@@ -7,15 +7,15 @@
             "sub_dir": "Vulkan-Headers",
             "build_dir": "Vulkan-Headers/build",
             "install_dir": "Vulkan-Headers/build/install",
-            "commit": "v1.3.302"
+            "commit": "v1.4.303"
```
Comment (spencer-lunarg) [inline scripts/known_good.json:10]:
```suggestion
            "commit": "6a74a7d65cafa19e38ec116651436cce6efd5b2e"
```

Should fix the tests now

Comment (arno-lunarg) [reply scripts/known_good.json:10]: Not even needed, the reference behind the 1.4.303 tag has been updated!

**Comment by arno-lunarg** [comment]
Ran the whole test suite with a Nvidia 1.4 driver. Had some issues but they seem driver related. One test needs to be updated, here is a patch:
[fix_test_vvl_1_4.patch](https://github.com/user-attachments/files/18009580/fix_test_vvl_1_4.patch)

**Comment by spencer-lunarg** [comment]
closed as https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8966 merged headers

**Pull request closed without merging**